### PR TITLE
tests: remove binary assumption, fixes #34

### LINKED ISSUE #34: Tests failing for Node 6
Looks to be an issue with `Buffer`s and `utf8`/`binary` again.
Did something change in Node 6 that is going to make us incompatible?

@feross any ideas?

### PR DESCRIPTION
**Pull request by dcousens**

### CODE REVIEW

**Comment by dcousens** [comment]
resolves #34,  doesn't add the warning though. 

Ping @dominictarr @indutny @calvinmetcalf and @fanatid on whether we should add that warning, reference: https://github.com/crypto-browserify/pbkdf2/issues/34#issuecomment-231659459

**Comment by dcousens** [comment]
Naturally,  this is going to start failing on Travis on all the pre-binary versions... I'm not sure what the best course of action is.

**Comment by fanatid** [comment]
Since this package for browser not whether we should follow by latest active LTS node release?
So, I prefer `utf8` by default, drop node 0.10 support and release major version after  2016-10-01.

**Comment by dcousens** [comment]
@fanatid the issue is,  Node 6 is the only that one that *isn't failing* after this PR

**Comment by fanatid** [comment]
> Node 6 is the only that one that isn't failing after this PR

this should fix the issue:
```js
if (!process.browser && pVersionMajor >= 6) runTests('node pbkdf2', require('../'))
```

@dcousens under drop 0.10 support I meant remove shim pbkdf2Sync

**Comment by dcousens** [comment]
@fanatid so we just curtail the valid test fixtures with UTF8 vs non-UTF8?
I can work around it,  but,  is that all we want to do?  Make the tests pass and move on?

I guess I'm ~OK with that,  but,  it just sounds like it could hurt someone.

**Comment by fanatid** [comment]
we also can use 2 sets of fixtures (utf8 and binary)

File: browser.js [hunk 1]
```diff
@@ -16,8 +16,8 @@ exports.pbkdf2 = function (password, salt, iterations, keylen, digest, callback)
 }
 
 exports.pbkdf2Sync = function (password, salt, iterations, keylen, digest) {
-  if (!Buffer.isBuffer(password)) password = new Buffer(password, 'binary')
-  if (!Buffer.isBuffer(salt)) salt = new Buffer(salt, 'binary')
+  if (!Buffer.isBuffer(password)) password = new Buffer(password, 'utf-8')
+  if (!Buffer.isBuffer(salt)) salt = new Buffer(salt, 'utf-8')
```
Comment (fanatid) [inline browser.js]: You don't need second argument here now, encoding is 'utf8' by default.
Maybe even use `Buffer.from(..)` ?

**Comment by indutny** [comment]
Other than one nit by @fanatid , LGTM

File: browser.js (hunk 1 above)
Comment (dcousens) [inline browser.js]: wouldn't `Buffer.from` completely break previous versions?

Comment (fanatid) [inline browser.js]: yea, sorry, `.from` will break all previous versions :(

File: browser.js [hunk 2]
```diff
@@ -16,8 +16,8 @@ exports.pbkdf2 = function (password, salt, iterations, keylen, digest, callback)
 }
 
 exports.pbkdf2Sync = function (password, salt, iterations, keylen, digest) {
-  if (!Buffer.isBuffer(password)) password = new Buffer(password, 'binary')
-  if (!Buffer.isBuffer(salt)) salt = new Buffer(salt, 'binary')
+  if (!Buffer.isBuffer(password)) password = new Buffer(password)
+  if (!Buffer.isBuffer(salt)) salt = new Buffer(salt)
```
Comment (dcousens) [inline browser.js]: @fanatid they are now defaulted

File: browser.js [hunk 3]
```diff
@@ -15,9 +16,11 @@ exports.pbkdf2 = function (password, salt, iterations, keylen, digest, callback)
   })
 }
 
+var defaultEncoding = (process.browser || pVersionMajor >= 6) ? 'utf-8' : 'binary'
+
 exports.pbkdf2Sync = function (password, salt, iterations, keylen, digest) {
-  if (!Buffer.isBuffer(password)) password = new Buffer(password, 'binary')
-  if (!Buffer.isBuffer(salt)) salt = new Buffer(salt, 'binary')
+  if (!Buffer.isBuffer(password)) password = new Buffer(password, defaultEncoding)
+  if (!Buffer.isBuffer(salt)) salt = new Buffer(salt, defaultEncoding)
```
Comment (dcousens) [inline browser.js]: gah,  this sucks.  any ideas @calvinmetcalf @fanatid @indutny ?

**Comment by dcousens** [comment]
@fanatid RE

> @dcousens under drop 0.10 support I meant remove shim pbkdf2Sync

can you elaborate?

File: browser.js (hunk 3 above)
Comment (indutny) [inline browser.js]: What exactly?

**Comment by fanatid** [comment]
>> @dcousens under drop 0.10 support I meant remove shim pbkdf2Sync

> can you elaborate?

after 2016-10-01 node 0.10 will not be supported, so why not remove shim?

**Comment by dcousens** [comment]
Right,  sorry,  not sure how that didn't read correctly. 
But that doesn't resolve the other issues, its just a way to remove code.
I'm happy with that :+1:

**Comment by dcousens** [comment]
@fanatid removed `0.10` support

File: browser.js (hunk 3 above)
Comment (dcousens) [inline browser.js]: Just the situation,  wondering if you had any thoughts on impl.
Otherwise,  I assume this PR is good to go now

**Comment by dcousens** [comment]
Ready to merge

**Comment by dcousens** [comment]
Published as `pbkdf2@3.0.5`

**Pull request merged**